docs/network-backends.md

```diff
@@ -1,3 +1,268 @@
+
+## Working with network backends
+
+### The default network backend
+
+Typically you won't need to specify a network backend, as a default will automatically be selected. However, understanding how the network backends fit in may be useful if you want to better understand the underlying architecture. Let's start by seeing how we can explicitly select the network backend.
+
+First we're making a standard HTTP request, using a connection pool:
+
+```python
+import httpcore
+
+with httpcore.ConnectionPool() as http:
+    response = http.request('GET', 'https://www.example.com')
+```
+
+We can also have the same behavior, but be explicit with our selection of the network backend:
+
+```python
+import httpcore
+
+network_backend = httpcore.NetworkBackend()
+with httpcore.ConnectionPool(network_backend=network_backend) as http:
+    response = http.request('GET', 'https://www.example.com')
+```
+
+The `httpcore.NetworkBackend()` implementation handles the opening of TCP connections, and operations on the socket stream, such as reading, writing, and closing the connection.
+
+We can get a better understanding of this by using a network backend to send a basic HTTP/1.1 request directly:
+
+```python
+import httpcore
+import ssl
+import certifi
+
+# Create an SSL context using 'certifi' for the certificates.
+ssl_context = ssl.create_default_context()
+ssl_context.load_verify_locations(certifi.where())
+
+# A basic HTTP/1.1 request as a plain bytestring.
+request = b''.join([
+    b'GET / HTTP/1.1\r\n',
+    b'Host: www.example.com\r\n',
+    b'Accept: */*\r\n',
+    b'Connection: close\r\n',
+    b'\r\n'
+])
+
+# Open a TCP stream and upgrade it to SSL.
+network_backend = httpcore.NetworkBackend()
+network_stream = network_backend.connect_tcp("www.example.com", 443)
+network_stream.start_tls(ssl_context, server_hostname="www.example.com")
+
+# Send the HTTP request.
+network_stream.write(request)
+
+# Read the HTTP response.
+while True:
+    response = network_stream.read(max_bytes=4096)
+    if response == b'':
+        break
+    print(response)
+```
+
+### Async network backends
+
+If we're working with an `async` codebase, then we need to select a different backend.
+
+The `httpcore.AnyIONetworkBackend` is suitable for usage if you're running under `asyncio`. This is a networking backend implemented using [the `anyio` package](https://anyio.readthedocs.io/en/3.x/).
+
+```python
+import httpcore
+import asyncio
+
+async def main():
+    network_backend = httpcore.AnyIONetworkBackend()
+    async with httpcore.AsyncConnectionPool(network_backend=network_backend) as http:
+        response = await http.request('GET', 'https://www.example.com')
+
+asyncio.run(main())
+```
+
+The `AnyIONetworkBackend` will work when running under either `asyncio` or `trio`. However, if you're working with async using the [`trio` framework](https://trio.readthedocs.io/en/stable/), then we recommend using the `httpcore.TrioNetworkBackend`.
```

Comment:
Is there a reason to answer why we recommend that? Because that question pop up in my head.

Reply:
Thanks. Added some text to describe this more clearly.